### PR TITLE
FIx #5569 add no_uwsgi control to setup.py

### LINKED ISSUE #5569: 20230316 uwsgi optional in setup.py
uwsgi is not required for sirepo. Make it optional with a command line argument or environment variable.

### PR DESCRIPTION
**Pull request by robnagler**

### CODE REVIEW

**Review by Benbenbenin0 — CHANGES_REQUESTED** [review]
Stylistic questions/changes

File: setup.py [hunk 1]
```diff
@@ -5,35 +5,40 @@
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
```
[the remark this answers is not in the document]
Comment (robnagler) [reply setup.py:5]: 461e62d6dd08e8895de8e4bb63bd57f18813525d but we don't usually do this. Copyright is so long that it's somewhat irrelevant.

File: setup.py [hunk 2]
```diff
@@ -5,35 +5,40 @@
...
 """
 import pykern.pksetup
+import os
```
[the remark this answers is not in the document]
Comment (robnagler) [reply setup.py:7]: 461e62d6dd08e8895de8e4bb63bd57f18813525d

File: setup.py [hunk 3]
```diff
@@ -5,35 +5,40 @@
...
-        # required for email login and smtp
-        "Authlib>=0.13",
-        "dnspython",
-        # required for sbatch agent
-        "asyncssh",
-    ],
+    install_requires=install_requires,
     license="http://www.apache.org/licenses/LICENSE-2.0.html",
     name="sirepo",
     url="http://sirepo.com",
```
[the remark this answers is not in the document]
Comment (robnagler) [reply setup.py:44]: Maybe I'm not understanding which newline you mean:

```txt
$ od -c setup.py | tail -2
0003300   "   ,  \n                   ]   ,  \n   )  \n
0003314
```

[the remark this answers is not in the document]
Comment (Benbenbenin0) [reply setup.py:44]: My mistake, Github editor didn't show EOF newline

**Review by Benbenbenin0 — APPROVED** [review]

**Pull request merged**